scripts/20210517_validation-object/script.js

```diff
@@ -0,0 +1,394 @@
+/* eslint-disable */
+// Scripts for #408
+
+/**
+ * Part 1: Verify that the database is internally consistent today
+ * for all short text, long text and number fields.
+ * Expect 0 records for all scripts
+ */
+
+// a. customVal === customMin when selectedValidation === 'Minimum'
+// Expect 0 records
+
+db.getCollection('forms').aggregate([
+  {
+    $match: {
+      'form_fields': {$elemMatch: {'fieldType': { $in: ['textfield', 'textarea', 'number'] }}},
+    },
+  },
+  { $unwind: '$form_fields' },
+  {
+    $match: {
+      'form_fields.fieldType': { $in: ['textfield', 'textarea', 'number'] },
+    },
+  },
+  {
+    $project: {
+      form_fields: 1,
+      selectedValidation: '$form_fields.ValidationOptions.selectedValidation',
+      customVal: '$form_fields.ValidationOptions.customVal',
+      customMin: '$form_fields.ValidationOptions.customMin',
+      customMax: '$form_fields.ValidationOptions.customMax',
+    },
+  },
+  {
+    $match: {
+      $and: [
+        {
+          selectedValidation: 'Minimum',
+        },
+        {
+          $expr: { $ne: ['$customMin', '$customVal'] },
```

Comment:
it would be good to sanity-check this locally by deliberately inserting a field where these are not equal and ensuring that the query finds the correct result